clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -4338,6 +4338,9 @@ def warn_param_typestate_mismatch : Warning<
 def warn_unknown_sanitizer_ignored : Warning<
   "unknown sanitizer '%0' ignored">, InGroup<UnknownSanitizers>;
 
+def warn_impcast_matrix_scalar : Warning<
```

Comment:
for both `warn_impcast_matrix_scalar ` and `warn_hlsl_impcast_matrix_truncation` I was thinking about doing %select{vector|matrix}0 and just merging with the vector versions but these warnings are consumed by `DiagnoseImpCast` and that would have been a big change impacting many  warnings and helper functions. So it was just easier to have two seperate matrix warnings.